slowapi/wrappers.py

```diff
@@ -82,8 +82,9 @@ def __iter__(self) -> Iterator[Limit]:
             if "key" in inspect.signature(self.__limit_provider).parameters.keys():
                 assert (
                     "request" in inspect.signature(self.key_function).parameters.keys()
-                )
-                assert self.request
+                ), "If limit provider function depends on `key` argument, key function must accept `request`"
```

Comment:
To make it easier for the developer to figure out what the problem is, you could do something like:
```suggestion
                ), f"Limit provider function {self.key_function.__name__} needs a `request` argument"
```